### DISCUSSION TITLE
Running into issues using script event triggers

### DISCUSSION THREAD

**Discussion by daranable** [TriliumNext/Trilium#1849] [Q&A] [answered]
Hi, just recently started using trilium, and loving it so far I am trying to start scripting a few automations for myself and trying to base my work on the weight tracker script.  However, I am trying to have a script trigger when an attribute is updated on my current day journal so I can store it's input into a JSON file.  So I tried adding the `~runOnAttributeChange=onDinnerChanged` to the day template, but it doesn't appear to get copied to the day when the day note is generated, and even when I manually put the event on the day note I still can't get the script to trigger when I use the box of the promoted attribute.

I next tried adding the event like this on the journal note at the origin of the tree like this:
`~child:child:child:runOnAttributeChange=onDinnerChanged `
Copying how the template is set for each day, still with no luck getting this event to trigger.

As of right now I haven't event started on the script all my script note's content is at this point is:
`api.showMessage('Hello, World!');`

So, I am not sure if I am missing something obvious or if I have found a bug, any help that you can offer would be greatly appreciated!

I am currently on version 0.46.7 build revision `ef37a52a06b471e60f9c0f11da704283bbcef6ab`

**Comment by zadam** [comment, marked as answer]
Hi, so what you write all sounds correct - so things should work, but don't which seems to suggest that something not described is wrong.

> As of right now I haven't event started on the script all my script note's content is at this point is:
> api.showMessage('Hello, World!');

This suggests that you've created "JS frontend" script and that might be the problem. Only "JS backend" scripts can listen to the entity (attribute) changes.

**Reply by daranable** [reply]
That would definitely be what I'm missing! Is there any documentation covering what events trigger backend vs frontend scripts? I don't recall that being documented on the events wiki page.

I was setting it up as a frontend script, I'll switch to backend then.  Thank you very much!

**Reply by zadam** [reply]
https://github.com/zadam/trilium/wiki/Events

> Note that these are backend events and thus relation need to point to the "JS backend" code note.

But it's of course easy to overlook ...

**Reply by daranable** [reply]
Haha, go figure! No matter how much I try to read things thoroughly I keep managing to miss those important details.
